### DISCUSSION TITLE
Double path to docs

### DISCUSSION THREAD

**Discussion by pow3rpi** [fastapi/fastapi#11751] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [ ] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [x] I commit to help with one of those options 👆

### Example Code

```python
import uvicorn
from fastapi import FastAPI, APIRouter

router = APIRouter(prefix='/v1')


@router.get('/healthcheck', tags=['Healthcheck'])
def check_service():
    return {'status': 'OK'}


def create_app() -> FastAPI:
    api = FastAPI(
        title='Test App',
        root_path='/api',
        root_path_in_servers=False,
    )
    api.include_router(router)
    return api


app = create_app()

if __name__ == '__main__':
    uvicorn.run(
        'test:app',
        host='0.0.0.0',
        port=5000,
        reload=True
    )
```

### Description

Actually, I am not sure if it is a FastAPI issue. It can be on the [Swagger UI](https://github.com/swagger-api/swagger-ui) side but I haven't found any related issues in their repository, so that's why I've decided to open the discussion here.

The problem is that when I use **root_path** parameter in FastAPI application I get documentation accessible on 2 paths: default **/docs** and **root_path+/docs**.

Currently, I see 2 ways how I can handle this issue:

1. For example, use a different value for **docs_url**, like:  **docs_url='/docs-api'**, which is something I am not satisfied with.
`api = FastAPI(title='Test App', root_path='/api', root_path_in_servers=False, docs_url='/docs-api')`
2. Change router prefix to **/api/v1**, remove **root_path** parameter and update **docs_url** with **/api/docs**. This is also not the best solution because each endpoint starts displaying **/api/v1** prefix in swagger.
`router = APIRouter(prefix='/api/v1')`
`api = FastAPI(title='Test App', root_path_in_servers=False, docs_url='/api/docs')`

Let me clarify that if I use:
`api = FastAPI(title='Test App', root_path='/api', root_path_in_servers=False, docs_url='/api/docs')`
then, swagger is accessible on very strange path **/api/api/docs**.

I reckon that it would be best to have opportunity to switch off default **/docs** path if we have **root_path** parameter specified. I don't insist that it should be a default setting but maybe we could control it via additional parameter.

Feel free to express your opinion on the topic.

### Operating System

Linux

### FastAPI Version

0.111.0

### Pydantic Version

1.10.10

### Python Version

3.11.9

**Comment by YuriiMotov** [comment, marked as answer]
When you use `root_path='/api'` parameter in `FastAPI()` it will make all your endpoint accessible via two paths: 1) normal (`http://localhost:5000/v1/healthcheck`) and 2) prefixed (`http://localhost:5000/api/v1/healthcheck`).

As I understand this is done this way to make the app compatible with different ASGI servers (not sure about this).

To make it accessible by only normal paths you can remove that parameter from `FastAPI()` and pass it to `uvicorn.run()` (you should [run your app behind the stripping path proxy](https://fastapi.tiangolo.com/advanced/behind-a-proxy/?h=trae#testing-locally-with-traefik), otherwise it will fail to show docs):

```python
import uvicorn
from fastapi import FastAPI, APIRouter

router = APIRouter(prefix='/v1')


@router.get('/healthcheck', tags=['Healthcheck'])
def check_service():
    return {'status': 'OK'}


def create_app() -> FastAPI:
    api = FastAPI(
        title='Test App',
        # root_path='/api',
        root_path_in_servers=False,
    )
    api.include_router(router)
    return api


app = create_app()

if __name__ == '__main__':
    uvicorn.run(
        'test:app',
        host='0.0.0.0',
        port=5000,
        root_path="/api",
        reload=True
    )
```

Now, you `/healthcheck` endpoint will only be accessible via `http://localhost:5000/v1/healthcheck` (if you try requesting it directly from ASGI server). And from outside the proxy the path will be `{host:port}/api/v1/healthcheck`
